src/sandstorm/run-bundle.c++

```diff
@@ -1861,17 +1873,55 @@ private:
       return false;
     }
 
-    // Start http request to download bundle.
+    // Download bundle to temporary file.
     auto url = kj::str("https://dl.sandstorm.io/sandstorm-", targetBuild, ".tar.xz");
+    auto file = openTemporary("/var/tmp/sandstorm-update");
     context.warning(kj::str("Downloading: ", url));
-    auto download = kj::heap<CurlRequest>(url);
-    int fd = download->getPipe();
-    unpackUpdate(fd, kj::mv(download), targetBuild);
+    CurlRequest(url, file);
+    KJ_SYSCALL(lseek(file, 0, SEEK_SET));
+
+    // Verify signature.
+    {
+      context.warning("Checking signature...");
+      KJ_ON_SCOPE_FAILURE(
+          context.warning("*** SIGNATURE CHECK FAILED: please notify security@sandstorm.io ***"));
```

Comment:
If I'm reading this right, this will complain loudly with a possibly-inaccurate message if simply downloading the signature file fails (think network timeout or 502 or something). I'm inclined to think that a brief network outage should be slightly less terrifying.  Maybe move the `CurlRequest` below out into the containing scope, and only complain if you successfully fetch the signature and then it fails verification?